xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.cpp

```diff
@@ -538,10 +513,27 @@ void CDVDDemuxClient::RequestStream(CDemuxStream *stream, std::map<int, std::sha
 
   // only update profile / level if we create a new stream
   // existing streams may be corrected by ParsePacket
+  CDemuxStream *extraSource(nullptr);
   if (!oldStream)
   {
     newStream->profile = stream->profile;
     newStream->level = stream->level;
+    extraSource = stream;
+  }
+  else if (oldStream != newStream)
+  {
+    newStream->profile = oldStream->profile;
+    newStream->level = oldStream->level;
+    extraSource = oldStream.get();
```

Comment:
this is kind of ugly and should not be required. if we create a new stream, we activate the parser that will split extradata. that will trigger a sream change event.